Makefile

```diff
@@ -20,22 +20,45 @@ lint:
 	exit
 
 .PHONY: build
-build:
+build: generate-prod
 	go build -tags 'production' $(LDFLAGS) -o $(BINPATH)/dp-frontend-articles-controller
 
 .PHONY: debug
-debug:
+debug: generate-debug
 	go build -tags 'debug' $(LDFLAGS) -o $(BINPATH)/dp-frontend-articles-controller
 	HUMAN_LOG=1 DEBUG=1 $(BINPATH)/dp-frontend-articles-controller
 
 .PHONY: test
-test:
-	go test -race -cover ./...
+test: generate-prod
+	go test -race -cover -tags 'production' ./...
 
 .PHONY: convey
 convey:
 	goconvey ./...
 
 .PHONY: test-component
 test-component:
-	go test -cover -coverpkg=github.com/ONSdigital/dp-frontend-articles-controller/... -component
\ No newline at end of file
+	go test -cover -coverpkg=github.com/ONSdigital/dp-frontend-articles-controller/... -component
+
+.PHONY: generate-debug
+generate-debug: fetch-dp-renderer
+	# fetch the renderer library and build the dev version
+	cd assets; go run github.com/kevinburke/go-bindata/go-bindata -prefix $(CORE_ASSETS_PATH)/assets -debug -o data.go -pkg assets locales/... templates/... $(CORE_ASSETS_PATH)/assets/locales/... $(CORE_ASSETS_PATH)/assets/templates/...
+	{ echo "// +build debug\n"; cat assets/data.go; } > assets/debug.go.new
+	mv assets/debug.go.new assets/data.go
+
+.PHONY: generate-prod
+generate-prod: fetch-dp-renderer
+	# fetch the renderer library and build the prod version
+	cd assets; go run github.com/kevinburke/go-bindata/go-bindata -prefix $(CORE_ASSETS_PATH)/assets -o data.go -pkg assets locales/... templates/... $(CORE_ASSETS_PATH)/assets/locales/... $(CORE_ASSETS_PATH)/assets/templates/...
+	{ echo "// +build production\n"; cat assets/data.go; } > assets/data.go.new
+	mv assets/data.go.new assets/data.go
+
+.PHONY: fetch-dp-renderer
+fetch-dp-renderer:
+ifeq ($(LOCAL_DP_RENDERER_IN_USE), 1)
+	$(eval CORE_ASSETS_PATH = $(shell grep -w "\"github.com/ONSdigital/dp-renderer\" =>" go.mod | awk -F '=> ' '{print $$2}' | tr -d '"'))
+else
+	$(eval APP_RENDERER_VERSION=$(shell grep "github.com/ONSdigital/dp-renderer" go.mod | cut -d ' ' -f2 ))
+	$(eval CORE_ASSETS_PATH = $(shell go get github.com/ONSdigital/dp-renderer@$(APP_RENDERER_VERSION) && go list -f '{{.Dir}}' -m github.com/ONSdigital/dp-renderer))
```

Comment:
is there is a need for these 2 lines - since the renderer is a dependancy then it must already have been downloaded so is this sufficient:
$(eval CORE_ASSETS_PATH = $(shell go list -f '{{.Dir}}' -m github.com/ONSdigital/dp-renderer))